packages/experimental/Avatar/SPEC.md

```diff
@@ -0,0 +1,311 @@
+
+### Initials
+
+Initials can be passed using `initials` prop or generated from `name` prop using getInitials method.
+Here are the requirements from Web spec:
+
+1. If only one word is provided, pull the first letter of the word.
+2. If 2 words provided, pull first letter of both words.
+3. If 2+ words provided, pull first letter of first word and last word.
+4. Ignore suffixes (i.e. Jr, III).
+5. Ignore words in parantheses Ignore special characters i.e. parantheses, dashes If numeric, fall back to Icon If first character is not alphabetic, fall back to icon.
+
+But requirements are still being discussed. This area will be updated once we come up with final solition. There are different implementations in different teams.
+
+### Name
+
+Name prop is used to generate initials.
+
+### Shape
+
+The Avatar supports a circular and square (with rounded corners) shape.
+
+### Sizes
+
+`Avatar` sopports next sizes: `20, 24, 28, 32, 36, 40, 48, 56, 64, 72, 96, 120`.
+Default size for win32 is 24.
+
+## API
+
+### Slots
+
+The `Avatar` component has the following slots:
+
+- `root` - The outer container representing the `Avatar` wrapper.
+- `image` - If specified, renders an `image`
+- `initials` - If specified, renders initials when there's no `image`
+- `icon` - If specified, renders an `icon` if there's no `image` or `initials`.
+- `initialsBackground` - It's a wrapper for initials and icon.
+- `ring` - If specified, renders the ring for the `Avatar` when active prop is set to `active`.
+- `badge` - If specified, renders presence badge. Example: `badge={{ status: 'outOfOffice' }}.`.
+
+The slots can be modified using the `compose` function on the `Avatar`. For more information on using the `compose` API, please see [this page](../../framework/composition/README.md).
+
+### Props
+
+```ts
+export interface JSAvatarProps extends IViewProps, AvatarConfigurableProps {
+  /**
+   * Optional activity indicator
+   * * active: the avatar will be decorated according to activeAppearance
+   * * inactive: the avatar will be reduced in size and partially transparent
+   * * unset: normal display
+   *
+   * @defaultvalue unset
+   */
+  active?: AvatarActive;
+
+  /**
+   * The appearance when `active="active"`
+   *
+   * @defaultvalue ring
+   */
+  activeAppearance?: AvatarActiveAppearance;
+
+  /**
+   * Badge shows the avatar's presence status.
+   * Badge can be shown only if `active` prop is undefined or unset
+   */
+  badge?: PresenceBadgeProps;
+
+  /**
+   * Icon to be displayed when the avatar doesn't have an image or initials.
+   *
+   * @defaultvalue `PersonRegular` (the default icon's size depends on the Avatar's size)
+   */
+  icon?: IconSourcesType;
+
+  /**
+   * Specify a string to be used instead of the name, to determine which color to use when color="colorful".
+   * Use this when a name is not available, but there is another unique identifier that can be used instead.
+   */
+  idForColor?: string | undefined;
```

Comment:
This feels quite specific and perhaps win32 only?

Reply:
I'm not sure if it's win32 only. We haven't implemented it yet. Web also has this prop. That's what spec says about this prop: "Specify a string to be used instead of the name, to determine which color to use when color="colorful". * Use this when a name is not available, but there is another unique identifier that can be used instead."